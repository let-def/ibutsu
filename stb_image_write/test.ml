let () = print_endline "Stb_image & Stb_image_write (u8)"

module Read = Stb_image
module Write = Stb_image_write

let test_write name image =
  let check filename = function
    | Result.Error (`Msg msg) ->
      Printf.printf "KO write %s: %s\n%!" filename msg;
      ""
    | Result.Ok () ->
      let digest = (Digest.file filename |> Digest.to_hex) in
      Printf.printf "OK %s: %s\n%!" filename digest;
      digest
  in
  let bmp =
    let filename = name ^ ".bmp" in
    check filename (Write.bmp filename image)
  in
  let png =
    let filename = name ^ ".png" in
    check filename (Write.png filename image)
  in
  let tga =
    let filename = name ^ ".tga" in
    check filename (Write.tga filename image)
  in
  let jpg =
    let filename = name ^ ".jpg" in
    check filename (Write.jpg filename image)
  in
  let jpg100 =
    let filename = name ^ "100.jpg" in
    check filename (Write.jpg ~quality:100 filename image)
  in
  let jpg50 =
    let filename = name ^ "50.jpg" in
    check filename (Write.jpg ~quality:50 filename image)
  in
  let vflip =
    let filename = name ^ "vflip.png" in
    Stb_image.vflip image;
    let result = check filename (Write.png filename image) in
    Stb_image.vflip image;
    result
  in
  let mipmap =
    let filename = name ^ "mipmap.png" in
    match
      let w = Image_type.width image in
      let h = Image_type.height image in
      let c = Image_type.channels image in
      Image_type.image
        ~width:(w / 2)
        ~height:(h / 2)
        ~channels:c
        ~offset:10
        ~stride:(w * c)
        (Bigarray.Array1.create
           Bigarray.int8_unsigned Bigarray.c_layout
           (10 + (w * h / 2) * c))
    with
    | Result.Error (`Msg msg) ->
      "failed to allocate mipmap buffer -- " ^ msg
    | Result.Ok image' ->
      Stb_image.mipmap image image';
      check filename (Write.png filename image')
  in
  let blur =
    let filename = name ^ "blur.png" in
    Stb_image.expblur image ~radius:2.0;
    let result = check filename (Write.png filename image) in
    result
  in
  [bmp; png; tga; jpg; jpg100; jpg50; vflip; mipmap; blur]

let check_and_write ?(and_then=ignore) name = function
  | Result.Error (`Msg msg) ->
    Printf.printf "KO read %s: %s\n%!" name msg;
    []
  | Result.Ok image ->
    let result = test_write name image in
    and_then image;
    result

let check_and_write ?and_then name load decode =
  let disk = check_and_write ?and_then ("disk_" ^ name) load in
  let mem = check_and_write ?and_then ("mem_" ^ name) decode in
  if (disk <> mem) then
    Printf.printf "KO: on disk and in memory decode differ for %s\n%!" name

let padded_image = function
  | Result.Error _ as error -> error
  | Result.Ok image ->
    let w = Image_type.width image in
    let h = Image_type.height image in
    let c = Image_type.channels image in
    let offset = w * 3 / 2 in
    let stride = (w + w * 1 / 3) * c in
    let buffer =
      Bigarray.Array1.create
        Bigarray.int8_unsigned Bigarray.c_layout
        (offset + stride * h * c)
    in
    match
      Image_type.image ~width:w ~height:h ~channels:c ~offset ~stride buffer
    with
    | Result.Error _ as error -> error
    | Result.Ok image' as result ->
      Stb_image.copy image image';
      result


let () = (
  let ic = open_in_bin "../test_pokeball.png" in
  let buffer =
    Unix.map_file (Unix.descr_of_in_channel ic)
      Bigarray.int8_unsigned Bigarray.c_layout false [|in_channel_length ic|]
    |> Bigarray.array1_of_genarray
  in
  check_and_write "pokeball"
    (Read.load "test_pokeball.png")
    (Read.decode buffer);
  check_and_write "pokeball_1"
    (Read.load ~channels:1 "../test_pokeball.png")
    (Read.decode ~channels:1 buffer);
  check_and_write "pokeball_2"
    (Read.load ~channels:2 "../test_pokeball.png")
    (Read.decode ~channels:2 buffer);
  check_and_write "pokeball_3"
    (Read.load ~channels:3 "../test_pokeball.png")
    (Read.decode ~channels:3 buffer);
  check_and_write "pokeball_4"
    (Read.load ~channels:4 "../test_pokeball.png")
    (Read.decode ~channels:4 buffer);
  check_and_write "pokeball_unmanaged"
    (Read.load_unmanaged "../test_pokeball.png")
    (Read.decode_unmanaged buffer)
    ~and_then:Read.free_unmanaged;
  check_and_write "pokeball_1_unmanaged"
    (Read.load_unmanaged ~channels:1 "../test_pokeball.png")
    (Read.decode_unmanaged ~channels:1 buffer)
    ~and_then:Read.free_unmanaged;
  check_and_write "pokeball_2_unmanaged"
    (Read.load_unmanaged ~channels:2 "../test_pokeball.png")
    (Read.decode_unmanaged ~channels:2 buffer)
    ~and_then:Read.free_unmanaged;
  check_and_write "pokeball_3_unmanaged"
    (Read.load_unmanaged ~channels:3 "../test_pokeball.png")
    (Read.decode_unmanaged ~channels:3 buffer)
    ~and_then:Read.free_unmanaged;
  check_and_write "pokeball_4_unmanaged"
    (Read.load_unmanaged ~channels:4 "../test_pokeball.png")
    (Read.decode_unmanaged ~channels:4 buffer)
    ~and_then:Read.free_unmanaged;
  check_and_write "padded_pokeball_1"
    (padded_image (Read.load ~channels:1 "../test_pokeball.png"))
    (padded_image (Read.decode ~channels:1 buffer));
  check_and_write "padded_pokeball_2"
    (padded_image (Read.load ~channels:2 "../test_pokeball.png"))
    (padded_image (Read.decode ~channels:2 buffer));
  check_and_write "padded_pokeball_3"
    (padded_image (Read.load ~channels:3 "../test_pokeball.png"))
    (padded_image (Read.decode ~channels:3 buffer));
  check_and_write "padded_pokeball_4"
    (padded_image (Read.load ~channels:4 "../test_pokeball.png"))
    (padded_image (Read.decode ~channels:4 buffer));
  close_in ic;
)
