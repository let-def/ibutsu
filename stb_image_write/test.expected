Stb_image & Stb_image_write (u8)
KO read disk_pokeball: can't fopen
OK mem_pokeball.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK mem_pokeball.png: 97037a672c6547fe85a599b6bed76131
OK mem_pokeball.tga: 7351da8bc8f5b6457382cd2ff5894312
OK mem_pokeball.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball50.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeballvflip.png: b8300d7c02264972c53358be4c81cf7b
OK mem_pokeballmipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK mem_pokeballblur.png: b8d7a576b8079eb6b09bc6a02f8b6680
KO: on disk and in memory decode differ for pokeball
OK disk_pokeball_1.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_pokeball_1.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK disk_pokeball_1.tga: 7e581e04c05d3d4666a240db9f03959b
OK disk_pokeball_1.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_pokeball_1100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_pokeball_150.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_pokeball_1vflip.png: 40051e4ed445cf1a63ac109831f4483a
OK disk_pokeball_1mipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK disk_pokeball_1blur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK mem_pokeball_1.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_pokeball_1.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK mem_pokeball_1.tga: 7e581e04c05d3d4666a240db9f03959b
OK mem_pokeball_1.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_pokeball_1100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_pokeball_150.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_pokeball_1vflip.png: 40051e4ed445cf1a63ac109831f4483a
OK mem_pokeball_1mipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK mem_pokeball_1blur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK disk_pokeball_2.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_pokeball_2.png: f380a01014bcde29dabacaa8d95ac997
OK disk_pokeball_2.tga: adfc0095f15aad4528fc3925fc122ea8
OK disk_pokeball_2.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_pokeball_2100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_pokeball_250.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_pokeball_2vflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK disk_pokeball_2mipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK disk_pokeball_2blur.png: 089269b11b42f58df73dc74895fb5fff
OK mem_pokeball_2.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_pokeball_2.png: f380a01014bcde29dabacaa8d95ac997
OK mem_pokeball_2.tga: adfc0095f15aad4528fc3925fc122ea8
OK mem_pokeball_2.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_pokeball_2100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_pokeball_250.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_pokeball_2vflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK mem_pokeball_2mipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK mem_pokeball_2blur.png: 089269b11b42f58df73dc74895fb5fff
OK disk_pokeball_3.bmp: d9183b7545302f43cb6c615779270ed2
OK disk_pokeball_3.png: 84474ee0f1d97a26b830f39659d2632b
OK disk_pokeball_3.tga: f12cc71abbd847ae0e8d285ebcb78214
OK disk_pokeball_3.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_pokeball_3100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_pokeball_350.jpg: 914df247d5d124377da6be7dac66829e
OK disk_pokeball_3vflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK disk_pokeball_3mipmap.png: 5cdbef832654b5923afbf82f951992a8
OK disk_pokeball_3blur.png: 014b29043caa4e73714837b734feda04
OK mem_pokeball_3.bmp: d9183b7545302f43cb6c615779270ed2
OK mem_pokeball_3.png: 84474ee0f1d97a26b830f39659d2632b
OK mem_pokeball_3.tga: f12cc71abbd847ae0e8d285ebcb78214
OK mem_pokeball_3.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball_3100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball_350.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeball_3vflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK mem_pokeball_3mipmap.png: 5cdbef832654b5923afbf82f951992a8
OK mem_pokeball_3blur.png: 014b29043caa4e73714837b734feda04
OK disk_pokeball_4.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK disk_pokeball_4.png: 97037a672c6547fe85a599b6bed76131
OK disk_pokeball_4.tga: 7351da8bc8f5b6457382cd2ff5894312
OK disk_pokeball_4.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_pokeball_4100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_pokeball_450.jpg: 914df247d5d124377da6be7dac66829e
OK disk_pokeball_4vflip.png: b8300d7c02264972c53358be4c81cf7b
OK disk_pokeball_4mipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK disk_pokeball_4blur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK mem_pokeball_4.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK mem_pokeball_4.png: 97037a672c6547fe85a599b6bed76131
OK mem_pokeball_4.tga: 7351da8bc8f5b6457382cd2ff5894312
OK mem_pokeball_4.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball_4100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball_450.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeball_4vflip.png: b8300d7c02264972c53358be4c81cf7b
OK mem_pokeball_4mipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK mem_pokeball_4blur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK disk_pokeball_unmanaged.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK disk_pokeball_unmanaged.png: 97037a672c6547fe85a599b6bed76131
OK disk_pokeball_unmanaged.tga: 7351da8bc8f5b6457382cd2ff5894312
OK disk_pokeball_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_pokeball_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_pokeball_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK disk_pokeball_unmanagedvflip.png: b8300d7c02264972c53358be4c81cf7b
OK disk_pokeball_unmanagedmipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK disk_pokeball_unmanagedblur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK mem_pokeball_unmanaged.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK mem_pokeball_unmanaged.png: 97037a672c6547fe85a599b6bed76131
OK mem_pokeball_unmanaged.tga: 7351da8bc8f5b6457382cd2ff5894312
OK mem_pokeball_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeball_unmanagedvflip.png: b8300d7c02264972c53358be4c81cf7b
OK mem_pokeball_unmanagedmipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK mem_pokeball_unmanagedblur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK disk_pokeball_1_unmanaged.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_pokeball_1_unmanaged.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK disk_pokeball_1_unmanaged.tga: 7e581e04c05d3d4666a240db9f03959b
OK disk_pokeball_1_unmanaged.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_pokeball_1_unmanaged100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_pokeball_1_unmanaged50.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_pokeball_1_unmanagedvflip.png: 40051e4ed445cf1a63ac109831f4483a
OK disk_pokeball_1_unmanagedmipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK disk_pokeball_1_unmanagedblur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK mem_pokeball_1_unmanaged.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_pokeball_1_unmanaged.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK mem_pokeball_1_unmanaged.tga: 7e581e04c05d3d4666a240db9f03959b
OK mem_pokeball_1_unmanaged.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_pokeball_1_unmanaged100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_pokeball_1_unmanaged50.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_pokeball_1_unmanagedvflip.png: 40051e4ed445cf1a63ac109831f4483a
OK mem_pokeball_1_unmanagedmipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK mem_pokeball_1_unmanagedblur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK disk_pokeball_2_unmanaged.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_pokeball_2_unmanaged.png: f380a01014bcde29dabacaa8d95ac997
OK disk_pokeball_2_unmanaged.tga: adfc0095f15aad4528fc3925fc122ea8
OK disk_pokeball_2_unmanaged.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_pokeball_2_unmanaged100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_pokeball_2_unmanaged50.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_pokeball_2_unmanagedvflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK disk_pokeball_2_unmanagedmipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK disk_pokeball_2_unmanagedblur.png: 089269b11b42f58df73dc74895fb5fff
OK mem_pokeball_2_unmanaged.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_pokeball_2_unmanaged.png: f380a01014bcde29dabacaa8d95ac997
OK mem_pokeball_2_unmanaged.tga: adfc0095f15aad4528fc3925fc122ea8
OK mem_pokeball_2_unmanaged.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_pokeball_2_unmanaged100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_pokeball_2_unmanaged50.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_pokeball_2_unmanagedvflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK mem_pokeball_2_unmanagedmipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK mem_pokeball_2_unmanagedblur.png: 089269b11b42f58df73dc74895fb5fff
OK disk_pokeball_3_unmanaged.bmp: d9183b7545302f43cb6c615779270ed2
OK disk_pokeball_3_unmanaged.png: 84474ee0f1d97a26b830f39659d2632b
OK disk_pokeball_3_unmanaged.tga: f12cc71abbd847ae0e8d285ebcb78214
OK disk_pokeball_3_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_pokeball_3_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_pokeball_3_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK disk_pokeball_3_unmanagedvflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK disk_pokeball_3_unmanagedmipmap.png: 5cdbef832654b5923afbf82f951992a8
OK disk_pokeball_3_unmanagedblur.png: 014b29043caa4e73714837b734feda04
OK mem_pokeball_3_unmanaged.bmp: d9183b7545302f43cb6c615779270ed2
OK mem_pokeball_3_unmanaged.png: 84474ee0f1d97a26b830f39659d2632b
OK mem_pokeball_3_unmanaged.tga: f12cc71abbd847ae0e8d285ebcb78214
OK mem_pokeball_3_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball_3_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball_3_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeball_3_unmanagedvflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK mem_pokeball_3_unmanagedmipmap.png: 5cdbef832654b5923afbf82f951992a8
OK mem_pokeball_3_unmanagedblur.png: 014b29043caa4e73714837b734feda04
OK disk_pokeball_4_unmanaged.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK disk_pokeball_4_unmanaged.png: 97037a672c6547fe85a599b6bed76131
OK disk_pokeball_4_unmanaged.tga: 7351da8bc8f5b6457382cd2ff5894312
OK disk_pokeball_4_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_pokeball_4_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_pokeball_4_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK disk_pokeball_4_unmanagedvflip.png: b8300d7c02264972c53358be4c81cf7b
OK disk_pokeball_4_unmanagedmipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK disk_pokeball_4_unmanagedblur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK mem_pokeball_4_unmanaged.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK mem_pokeball_4_unmanaged.png: 97037a672c6547fe85a599b6bed76131
OK mem_pokeball_4_unmanaged.tga: 7351da8bc8f5b6457382cd2ff5894312
OK mem_pokeball_4_unmanaged.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_pokeball_4_unmanaged100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_pokeball_4_unmanaged50.jpg: 914df247d5d124377da6be7dac66829e
OK mem_pokeball_4_unmanagedvflip.png: b8300d7c02264972c53358be4c81cf7b
OK mem_pokeball_4_unmanagedmipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK mem_pokeball_4_unmanagedblur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK disk_padded_pokeball_1.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_padded_pokeball_1.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK disk_padded_pokeball_1.tga: 7e581e04c05d3d4666a240db9f03959b
OK disk_padded_pokeball_1.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_padded_pokeball_1100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_padded_pokeball_150.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_padded_pokeball_1vflip.png: 40051e4ed445cf1a63ac109831f4483a
OK disk_padded_pokeball_1mipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK disk_padded_pokeball_1blur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK mem_padded_pokeball_1.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_padded_pokeball_1.png: acf9dd9dd47e2f472d05c0d1a99849d8
OK mem_padded_pokeball_1.tga: 7e581e04c05d3d4666a240db9f03959b
OK mem_padded_pokeball_1.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_padded_pokeball_1100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_padded_pokeball_150.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_padded_pokeball_1vflip.png: 40051e4ed445cf1a63ac109831f4483a
OK mem_padded_pokeball_1mipmap.png: 3e790ec7de814acd76cdb4de25a85781
OK mem_padded_pokeball_1blur.png: 9e225b1e9c0eb1a0d53947a225f07e02
OK disk_padded_pokeball_2.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK disk_padded_pokeball_2.png: f380a01014bcde29dabacaa8d95ac997
OK disk_padded_pokeball_2.tga: adfc0095f15aad4528fc3925fc122ea8
OK disk_padded_pokeball_2.jpg: dae161cb898c3ec575f6916a42fe164e
OK disk_padded_pokeball_2100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK disk_padded_pokeball_250.jpg: 5d525059331620d0f42b3c520188f0d8
OK disk_padded_pokeball_2vflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK disk_padded_pokeball_2mipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK disk_padded_pokeball_2blur.png: 089269b11b42f58df73dc74895fb5fff
OK mem_padded_pokeball_2.bmp: 07ca445357a2bbfc93c7530dc0f9342f
OK mem_padded_pokeball_2.png: f380a01014bcde29dabacaa8d95ac997
OK mem_padded_pokeball_2.tga: adfc0095f15aad4528fc3925fc122ea8
OK mem_padded_pokeball_2.jpg: dae161cb898c3ec575f6916a42fe164e
OK mem_padded_pokeball_2100.jpg: be5e4c4b7e619e14e8c96fbaf36b1a58
OK mem_padded_pokeball_250.jpg: 5d525059331620d0f42b3c520188f0d8
OK mem_padded_pokeball_2vflip.png: 682f43d595e6c8149fcd0a1eadd94151
OK mem_padded_pokeball_2mipmap.png: cae67f00226fe4bd3a02f081a9b67562
OK mem_padded_pokeball_2blur.png: 089269b11b42f58df73dc74895fb5fff
OK disk_padded_pokeball_3.bmp: d9183b7545302f43cb6c615779270ed2
OK disk_padded_pokeball_3.png: 84474ee0f1d97a26b830f39659d2632b
OK disk_padded_pokeball_3.tga: f12cc71abbd847ae0e8d285ebcb78214
OK disk_padded_pokeball_3.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_padded_pokeball_3100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_padded_pokeball_350.jpg: 914df247d5d124377da6be7dac66829e
OK disk_padded_pokeball_3vflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK disk_padded_pokeball_3mipmap.png: 5cdbef832654b5923afbf82f951992a8
OK disk_padded_pokeball_3blur.png: 014b29043caa4e73714837b734feda04
OK mem_padded_pokeball_3.bmp: d9183b7545302f43cb6c615779270ed2
OK mem_padded_pokeball_3.png: 84474ee0f1d97a26b830f39659d2632b
OK mem_padded_pokeball_3.tga: f12cc71abbd847ae0e8d285ebcb78214
OK mem_padded_pokeball_3.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_padded_pokeball_3100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_padded_pokeball_350.jpg: 914df247d5d124377da6be7dac66829e
OK mem_padded_pokeball_3vflip.png: 5d6d5afd6e26e5ccf2d01a1c0b9a76b3
OK mem_padded_pokeball_3mipmap.png: 5cdbef832654b5923afbf82f951992a8
OK mem_padded_pokeball_3blur.png: 014b29043caa4e73714837b734feda04
OK disk_padded_pokeball_4.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK disk_padded_pokeball_4.png: 97037a672c6547fe85a599b6bed76131
OK disk_padded_pokeball_4.tga: 7351da8bc8f5b6457382cd2ff5894312
OK disk_padded_pokeball_4.jpg: 7261a4aa08679de4896ac9b212129864
OK disk_padded_pokeball_4100.jpg: de951ca8f880a20e638994eea3c07cf4
OK disk_padded_pokeball_450.jpg: 914df247d5d124377da6be7dac66829e
OK disk_padded_pokeball_4vflip.png: b8300d7c02264972c53358be4c81cf7b
OK disk_padded_pokeball_4mipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK disk_padded_pokeball_4blur.png: b8d7a576b8079eb6b09bc6a02f8b6680
OK mem_padded_pokeball_4.bmp: ae94d8fe916c02ff4870087ca8313ee2
OK mem_padded_pokeball_4.png: 97037a672c6547fe85a599b6bed76131
OK mem_padded_pokeball_4.tga: 7351da8bc8f5b6457382cd2ff5894312
OK mem_padded_pokeball_4.jpg: 7261a4aa08679de4896ac9b212129864
OK mem_padded_pokeball_4100.jpg: de951ca8f880a20e638994eea3c07cf4
OK mem_padded_pokeball_450.jpg: 914df247d5d124377da6be7dac66829e
OK mem_padded_pokeball_4vflip.png: b8300d7c02264972c53358be4c81cf7b
OK mem_padded_pokeball_4mipmap.png: 1ad2c19b4ec0ecf7c524d1b8868fa834
OK mem_padded_pokeball_4blur.png: b8d7a576b8079eb6b09bc6a02f8b6680
